src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts

```diff
@@ -334,17 +334,64 @@ export class NotebookQuickPickProvider implements IQuickPickDataSource<OutlineEn
 
 export class NotebookOutlinePaneProvider implements IDataSource<NotebookCellOutline, OutlineEntry> {
 	constructor(
-		private _getEntries: () => OutlineEntry[],
+		private readonly outlineDataSourceRef: IReference<INotebookCellOutlineDataSource> | undefined,
 		@IConfigurationService private readonly _configurationService: IConfigurationService,
 	) { }
 
+	public getActiveEntry(): OutlineEntry | undefined {
+		const newActive = this.outlineDataSourceRef?.object?.activeElement;
+		if (!newActive) {
+			return undefined;
+		}
+
+		if (!this.filterEntry(newActive)) {
+			return newActive;
+		}
+
+		// find a valid parent
+		let parent = newActive.parent;
+		while (parent) {
+			if (this.filterEntry(parent)) {
```

Comment:
Some potential perf optimization:

- we are reading 3 settings each time when we read `filterEntry`, they can probably be cached as they won't change in this while loop.
- minor: maybe we can cache the active entry so we don't need to recompute if `activeElement` doesn't change.